### PR TITLE
API tweaks

### PR DESCRIPTION
**Pull request by dgasmith**
The PR does the following:

- Reworks imports to be more canonical.
- Runs a static analysis of the code base, changed quite a few little issues.
- Condenses API code, reworks several names, formats the code, and adds docstrings.
- Speeds up several tests using simpler methods and less dense 
- I have also fixed an issue with QCEngine where it would oversubscribe cores on VM's.

As a note, the examples are very large and you are using ~100MB on GitHub already. There is a cap of ~1GB. I would be hesitant to add additional large files as you cannot remove them from the repo due to history very easily.

### CODE REVIEW

File: crank/QMEngine.py [hunk 1]
```diff
@@ -69,18 +72,22 @@ def load_task_result_m(self, job_path=None):
...
         else:
             m = self.load_geomeTRIC_output()
         os.chdir(orig_dir)
         return m
 
-    def run(self, cmd, input_files=[], output_files=[]):
+    def run(self, cmd, input_files=None, output_files=None):
         """ Execute a command locally or remotely, based on whether self.work_queue is set """
-        if self.work_queue == None:
+        if input_files is None:
```
Comment (yudongqiu) [inline crank/QMEngine.py:86]: I'm not exactly sure why this change is needed.

Comment (dgasmith) [reply crank/QMEngine.py:86]: https://stackoverflow.com/questions/14247373/python-none-comparison-should-i-use-is-or

In general it doesn't matter too much; however, using `is` is the correct way according to the PEP and can save you some odd cases and issues if you are not careful down the line.

File: crank/crankServerAPI.py [hunk 2]
```diff
@@ -1,33 +0,0 @@
-# crankServerAPI.py
```
Comment (yudongqiu) [inline crank/crankServerAPI.py:1]: I originally separated the ServerAPI with crankAPI, because only the functions defined in serverAPI module need to be loaded and run by the server, while the functions in crankAPI are expected to be running on a "computing node" or "container"

File: crank/QMEngine.py (hunk 1 above)
Comment (yudongqiu) [reply crank/QMEngine.py:86]: Thanks, but my question about this part is: here 4 additional lines of code are added, to replace the default parameters `input_files=[]` to `input_files=None`, and overwrite it back.

**Comment by yudongqiu** [comment]
Thanks a lot for making this pull request. I have read through the changes, and I think reducing the cost of the tests is great, and it's completely fine to rename the parts that are mainly written for the server to use. Other than that, I have two mild suggestions about the code changes: first is that for the comment `changed quite a few little issues`, it would be great if you could provide a few comments in-place showing what are those issues; second is that there are a large amount of changes adjusting the "personal flavor" of the code, which includes moving imports up and down, adding empty lines and spaces, change `a == False` into `not a`, etc. These changes carry no "real improvements" (correct me if I'm wrong), but make the code base adopt some personal style. I'm fine with this PR but I feel these types of changes could dilute the "real improvements", and could make the code less familiar to the previous developers.

File: .gitignore [hunk 3]
```diff
@@ -105,3 +105,9 @@ ENV/
 
 # pytest
 .pytest_cache/
+
+# Test data
+crank/tests/files/
```
Comment (yudongqiu) [inline .gitignore:110]: This test folder include a few input files for the test_stack to run, so I think it shouldn't be ignored here.

File: crank/QMEngine.py (hunk 1 above)
Comment (dgasmith) [reply crank/QMEngine.py:86]: Oh declaring a list in the definition like that means you use the same list every time the function is invoked, effectively a global for the function. This can lead to very surprising things as your function now has state.

File: .gitignore (hunk 3 above)
Comment (dgasmith) [reply .gitignore:110]: You can still manually add more files here, but this will prevent accidentally adding many scratch files.

File: crank/crankServerAPI.py (hunk 2 above)
Comment (dgasmith) [reply crank/crankServerAPI.py:1]: You missed the get next jobs function I believe, either way it would be good to have the entire API in one place to make things a bit simpler IMO.

**Comment by dgasmith** [comment]
As a note I did not add personal flavor to the code, I moved the code to the canonically accepted way of writing these expressions and imports. 

Adding blank lines is probably the only personal style I added and I believe enhances readability. I can remove these if you wish.

File: crank/crankServerAPI.py (hunk 2 above)
Comment (yudongqiu) [reply crank/crankServerAPI.py:1]: The `crankAPI.get_next_jobs()` function is part of the crankAPI, which does a lot of computation and I think should run in the "crank computing node" or "crank container". The functions I put in crankServerAPI are the ones that need to run on the server, and I made them as light-weight as possible. Maybe I had a different understanding of how the workflow runs.
